cloud/amazon/ec2_vpc_dhcp_options.py

```diff
@@ -128,105 +157,204 @@
+        if to_delete and not add_only:
+            vpc_conn.delete_tags(resource_id, to_delete, dry_run=check_mode)
+
+        to_add = dict((k, tags[k]) for k in tags if k not in cur_tags)
+        if to_add:
+            vpc_conn.create_tags(resource_id, to_add, dry_run=check_mode)
+
+        latest_tags = get_resource_tags(vpc_conn, resource_id)
+        return {'changed': True, 'tags': latest_tags}
+    except EC2ResponseError as e:
+        raise Exception('Unable to update tags for {0}, error: {1}'.format(resource_id, e))
+
+def fetch_dhcp_options_for_vpc(vpc_conn, vpc_id):
     """
     Returns the DHCP options object currently associated with the requested VPC ID using the VPC
     connection variable.
     """
-    vpcs = vpc_connection.get_all_vpcs(vpc_ids=[vpc_id])
+    vpcs = vpc_conn.get_all_vpcs(vpc_ids=[vpc_id])
     if len(vpcs) != 1:
-      return None
-    dhcp_options = vpc_connection.get_all_dhcp_options(dhcp_options_ids=[vpcs[0].dhcp_options_id])
+        return None
+    dhcp_options = vpc_conn.get_all_dhcp_options(dhcp_options_ids=[vpcs[0].dhcp_options_id])
     if len(dhcp_options) != 1:
-      return None
+        return None
     return dhcp_options[0]
 
+def match_dhcp_options(vpc_conn, tags=None, options=None):
+    """
+    Finds a DHCP Options object that optionally matches the tags and options provided
+    """
+    dhcp_options = vpc_conn.get_all_dhcp_options()
+    for dopts in dhcp_options:
+        if tags and get_resource_tags(vpc_conn, dopts.id) == tags:
+            if options and dopts.options == options:
+                return(True, dopts)
+    return(False, None)
 
-def _get_vpcs_by_dhcp_options(dhcp_options_id, vpc_connection):
-    return vpc_connection.get_all_vpcs(filters={'dhcpOptionsId': dhcp_options_id})
-
-
-def _get_updated_option(requested, existing, inherit):
-    if inherit and (not requested or requested == ['']):
-        return existing
+def remove_dhcp_options_by_id(vpc_conn, dhcp_options_id):
+    associations = vpc_conn.get_all_vpcs(filters={'dhcpOptionsId': dhcp_options_id})
+    if len(associations) > 0:
+        return False
     else:
-        return requested
-
+        vpc_conn.delete_dhcp_options(dhcp_options_id)
+        return True
 
 def main():
     argument_spec = ec2_argument_spec()
     argument_spec.update(dict(
-        domain_name=dict(type='str', default=''),
-        dns_servers=dict(type='list', default=[]),
-        ntp_servers=dict(type='list', default=[]),
-        netbios_name_servers=dict(type='list', default=[]),
-        netbios_node_type=dict(type='int', default=2),
-        vpc_id=dict(type='str', required=True),
+        dhcp_options_id=dict(type='str', default=None),
+        domain_name=dict(type='str', default=None),
+        dns_servers=dict(type='list', default=None),
+        ntp_servers=dict(type='list', default=None),
+        netbios_name_servers=dict(type='list', default=None),
```

Comment:
Since you're changing the default from `[]` to `None`, can you update the `DOCUMENTATION` string to reflect the updated default, for dns_servers, ntp_servers, and netbios_name_servers?